src/app/views/media/clusters.jsx

```diff
@@ -31,10 +44,105 @@ const UserList3 = () => {
-        if (resp.status == 200) setClusters(resp.data)
-    }, [rowsPerPage, page]);
+    const handleSearch = (value) => {
+        setSearch(value); 
+        perfomSearch(value);
+    };
+
+    const languages = [
+        {label: "Español", value: "es"},
+        {label: "English", value: "us"},
+    ]
+
+    const handleLanguageByFilter = (event) =>{
+        const lang = event.target.name
+        const isChecked = event.target.checked
+
+        const updatedLangs = isChecked ? [...selectedLangs, lang] : selectedLangs.filter((selectedLangs) => selectedLangs !== lang)
+
+        setSelectedLangs(updatedLangs)
+
+        history.replace(
+            `/media/seo/cluster?${Object.keys({
+                limit: rowsPerPage,
+                offset: page * rowsPerPage,
+                like: search,
+                lang: updatedLangs.join(",")
+            })
+                .map(
+                  (key) => `${key}=${{ limit: rowsPerPage, offset: page * rowsPerPage, like: search, lang:updatedLangs.join(",") }[key]}`)
+                .join('&')}`
+            );
+        bc.registry()
+                .getAllClusters({ limit: rowsPerPage, 
+                    offset: page * rowsPerPage, 
+                    like: search, 
+                    lang: updatedLangs.join(",")
+                })
+                .then((res) =>{
+                setClusters(res.data)
+            });
+    }
+
+    const perfomSearch = useCallback(
+        debounce((search) => {
+            bc.registry()
+                .getAllClusters({ limit: rowsPerPage, 
+                    offset: page * rowsPerPage, 
+                    like: search, 
+                    lang: selectedLangs.join(",")
+                })
+                .then((res) =>{
+                setClusters(res.data)
+            });
+            history.replace(
+            `/media/seo/cluster?${Object.keys({
+                limit: rowsPerPage,
+                offset: page * rowsPerPage,
+                like: search,
+                lang: selectedLangs.join(",")
+            })
+                .map(
+                  (key) => `${key}=${{ limit: rowsPerPage, offset: page * rowsPerPage, like: search, lang:selectedLangs.join(",") }[key]}`)
+                .join('&')}`
+            );
+        }, 300),
+        [rowsPerPage, page, history, selectedLangs]
+    );
+
+    useEffect(() => {
+        const fetchClusters = async () => {
+            const resp = await bc.registry().getAllClusters({ 
+                limit: rowsPerPage, 
+                offset: page * rowsPerPage, 
+                like: search,
+                lang: selectedLangs.join(",") 
+
+            });
+            if (resp.status == 200) {
+                setClusters(resp.data);
+            }
+        };
+        fetchClusters();
+    }, [rowsPerPage, page, selectedLangs]);
+
+    const renderClusters = (clusters) => {
```

Comment:
You can use ternary conditions because we only have one condition. This render function you created is a good practice but it should only be used when having nested ternary conditions.